### PR TITLE
fix dplr: support initializing from a restart file

### PR DESCRIPTION
**Pull request by Yi-FanLi**
Thie PR aims at fixing #3679.

The `get_valid_pairs` will raise an error when
1) reading a restart file written from previous dplr jobs
2) reading a data file written from previous dplr jobs
3) executing 'run 0' for twice

The origin is that the check in `get_valid_pairs` is too strict for the initialization process. The check requires that a pair of real and virtual atoms found by iterating over the bondlist to be the local atoms of the same processor. This requirement is automatically satisfied during the running of MD because the positions of virtual atoms are set equal to those of real atoms before rebuilding bondlists. However, during initialization (`Verlet::setup(int flag)`), the bondlist is built based on the initial configuration provided by the user. If the user provides a configuration in which the virtual atoms are placed at the Wannier centroids, it is possible that a bond formed by a pair of real and virtual atoms are between a local and a ghost atom, which triggers this error.

This PR loosens the requirement on the initial configuration provided by the user. In the `setup_post_neighbor` function, the `get_valid_pairs` function searches for bonded pairs but skips the aforementioned check. It sets the coordinates of the virtual atoms to the coordinates of the real atoms, and then rebuilds the bondlist. Afterwards, the check can be passed. 

## Summary by CodeRabbit

- **New Features**
	- Introduced a new method for setting up pre-exchange operations, enhancing simulation lifecycle management.
- **Changes**
	- Renamed the `post_integrate` method to `pre_exchange` to better reflect its role in the simulation process.
	- Updated mask settings to improve clarity and context within the simulation lifecycle.
- **Chores**
	- Updated ownership context for Docker image in the build workflow configuration.

### CODE REVIEW

File: source/lmp/fix_dplr.cpp [hunk 1]
```diff
@@ -312,16 +313,68 @@ void FixDPLR::init() {
...
 /* ---------------------------------------------------------------------- */
 
 void FixDPLR::setup(int vflag) {
-  // if (strstr(update->integrate_style,"verlet"))
   post_force(vflag);
-  // else {
-  //   error->all(FLERR, "respa is not supported by this fix");
-  // }
+  printf("end of setup\n");
+  printf("nlocal = %d\n", atom->nlocal);
```
Comment (njzjz) [inline source/lmp/fix_dplr.cpp:377]: should be cleaned up

File: source/lmp/fix_dplr.cpp
Comment (njzjz) [inline source/lmp/fix_dplr.cpp:1]: could you add a test? For example, run 0 after run 0

Comment (Yi-FanLi) [reply source/lmp/fix_dplr.cpp:1]: Test added.

**Review by njzjz — APPROVED** [review]

File: source/lmp/fix_dplr.cpp [hunk 2]
```diff
@@ -312,25 +303,61 @@ void FixDPLR::init() {
...
+void FixDPLR::setup(int vflag) { post_force(vflag); }
 
 /* ---------------------------------------------------------------------- */
 
 void FixDPLR::min_setup(int vflag) { setup(vflag); }
 
 /* ---------------------------------------------------------------------- */
 
-void FixDPLR::get_valid_pairs(vector<pair<int, int> > &pairs) {
+void FixDPLR::get_valid_pairs(vector<pair<int, int> > &pairs, int setup) {
```
Comment (wanghan-iapcm) [inline source/lmp/fix_dplr.cpp:360]: could you plz change to name to e.g. `is_setup` or `strict_check_valid_bond_pair`, and change the type to bool.

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

File: source/lmp/fix_dplr.cpp (hunk 2 above)
Comment (Yi-FanLi) [reply source/lmp/fix_dplr.cpp]: Done!

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**